target/linux/pistachio/modules.mk

```diff
@@ -0,0 +1,236 @@
+endef
+
+$(eval $(call KernelPackage,nhc_dest))
+
+
+define KernelPackage/nhc_fragment
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=6LoWPAN IPv6 Fragment Header compression
+  DEPENDS:=@!LINUX_3_18 +kmod-6lowpan
+  KCONFIG:= \
+    CONFIG_6LOWPAN_NHC_FRAGMENT \
+    CONFIG_6LOWPAN_NHC=y
+  FILES:=$(LINUX_DIR)/net/6lowpan/nhc_fragment.ko
+  AUTOLOAD:=$(call AutoProbe,nhc_fragment)
+endef
+
+define KernelPackage/nhc_fragment/description
+  6LoWPAN IPv6 Fragment Header compression
+endef
+
+$(eval $(call KernelPackage,nhc_fragment))
+
+
+define KernelPackage/nhc_hop
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=6LoWPAN IPv6 Hop-by-Hop Options Header compression
+  DEPENDS:=@!LINUX_3_18 +kmod-6lowpan
+  KCONFIG:= \
+    CONFIG_6LOWPAN_NHC_HOP \
+    CONFIG_6LOWPAN_NHC=y
+  FILES:=$(LINUX_DIR)/net/6lowpan/nhc_hop.ko
+  AUTOLOAD:=$(call AutoProbe,nhc_hop)
+endef
+
+define KernelPackage/nhc_hop/description
+  6LoWPAN IPv6 Hop-by-Hop Options Header compression
+endef
+
+$(eval $(call KernelPackage,nhc_hop))
+
+
+define KernelPackage/nhc_ipv6
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=6LoWPAN IPv6 Header compression
+  DEPENDS:=@!LINUX_3_18 +kmod-6lowpan
+  KCONFIG:= \
+    CONFIG_6LOWPAN_NHC_IPV6 \
+    CONFIG_6LOWPAN_NHC=y
+  FILES:=$(LINUX_DIR)/net/6lowpan/nhc_ipv6.ko
+  AUTOLOAD:=$(call AutoProbe,nhc_ipv6)
+endef
+
+define KernelPackage/nhc_ipv6/description
+  6LoWPAN IPv6 Header compression
+endef
+
+$(eval $(call KernelPackage,nhc_ipv6))
+
+
+define KernelPackage/nhc_mobility
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=6LoWPAN IPv6 Mobility Header compression
+  DEPENDS:=@!LINUX_3_18 +kmod-6lowpan
+  KCONFIG:= \
+    CONFIG_6LOWPAN_NHC_MOBILITY \
+    CONFIG_6LOWPAN_NHC=y
+  FILES:=$(LINUX_DIR)/net/6lowpan/nhc_mobility.ko
+  AUTOLOAD:=$(call AutoProbe,nhc_mobility)
+endef
+
+define KernelPackage/nhc_mobility/description
+  6LoWPAN IPv6 Mobility Header compression
+endef
+
+$(eval $(call KernelPackage,nhc_mobility))
+
+
+define KernelPackage/nhc_routing
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=LoWPAN IPv6 Routing Header compression
+  DEPENDS:=@!LINUX_3_18 +kmod-6lowpan
+  KCONFIG:= \
+    CONFIG_6LOWPAN_NHC_ROUTING \
+    CONFIG_6LOWPAN_NHC=y
+  FILES:=$(LINUX_DIR)/net/6lowpan/nhc_routing.ko
+  AUTOLOAD:=$(call AutoProbe,nhc_routing)
+endef
+
+define KernelPackage/nhc_routing/description
+  LoWPAN IPv6 Routing Header compression
+endef
+
+$(eval $(call KernelPackage,nhc_routing))
+
+
+define KernelPackage/nhc_udp
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=6LoWPAN IPv6 UDP compression
+  DEPENDS:=@!LINUX_3_18 +kmod-6lowpan
+  KCONFIG:= \
+    CONFIG_6LOWPAN_NHC_UDP \
+    CONFIG_6LOWPAN_NHC=y
+  FILES:=$(LINUX_DIR)/net/6lowpan/nhc_udp.ko
+  AUTOLOAD:=$(call AutoProbe,nhc_udp)
+endef
+
+define KernelPackage/nhc_fragment/description
+  6LoWPAN IPv6 UDP compression
+endef
+
+$(eval $(call KernelPackage,nhc_udp))
+
+define KernelPackage/uccp420wlan
+  SUBMENU:=$(WIRELESS_MENU)
+  TITLE:=Imagination wireless driver
+  KCONFIG:=\
+    CONFIG_UCCP420WLAN \
+    CONFIG_CFG80211=y \
```

Comment:
The wifi driver should better use package/kernel/mac80211/ It is only possible to have one mac80211 and cfg80211 subsystem loaded into the kernel at the time and OpenWrt uses mac80211 from backports in package/kernel/mac80211/ . You can either add some patches to package/kernel/mac80211/ or add a separate package like  package/kernel/mwlwifi/Makefile .